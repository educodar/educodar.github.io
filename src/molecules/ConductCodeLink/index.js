import React from "react"
import { Link, withPrefix } from "gatsby"

const ConductCodeLink = ({ openInNew, ...props }) => {
  const Component = openInNew ? "a" : Link
  const path = "/codigo-de-conduta"
  props = {
    href: openInNew ? withPrefix(path) : null,
    rel: openInNew ? "noopener noreferrer" : null,
    target: openInNew ? "_blank" : "_self",
    to: openInNew ? null : path,
    ...props,
  }
  return <Component {...props}>CÃ³digo&nbsp;de&nbsp;Conduta</Component>
}

export default ConductCodeLink
