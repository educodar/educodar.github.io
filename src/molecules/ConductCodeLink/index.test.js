import React from "react"
import renderer from "react-test-renderer"
import ConductCodeLink from "."

describe("Conduct Code Link component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ConductCodeLink />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
