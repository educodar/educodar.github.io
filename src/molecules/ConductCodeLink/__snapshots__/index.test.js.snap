// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conduct Code Link component renders correctly 1`] = `
<a
  href="/codigo-de-conduta"
  rel={null}
  target="_self"
>
  Código de Conduta
</a>
`;
